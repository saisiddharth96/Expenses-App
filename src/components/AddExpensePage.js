import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm.js";
import { addExpense } from "../actions/expenses.js";

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.onSubmit(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <p>This is from Expense Page</p>
        <ExpenseForm buttonName="Add Expense" onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  onSubmit: expense => dispatch(addExpense(expense))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
