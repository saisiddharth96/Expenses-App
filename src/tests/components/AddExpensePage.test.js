import React from "react";
import { shallow } from "enzyme";
import { AddExpensePage } from "../../components/AddExpensePage.js";
import expenses from '../fixtures/expenses.js'

test("should render add expense page", () => {
  const onSubmit = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(
    <AddExpensePage onsubmit={onSubmit} history={history} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  const onSubmit = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(
    <AddExpensePage onsubmit={onSubmit} history={history} buttonName={"Add Expense"} />
  );
  wrapper.find("ExpenseForm").prop("onSubmit")(expenses[1]);

  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
})
