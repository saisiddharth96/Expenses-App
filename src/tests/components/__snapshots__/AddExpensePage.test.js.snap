// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render add expense page 1`] = `
<div>
  <p>
    This is from Expense Page
  </p>
  <ExpenseForm
    buttonName="Add Expense"
    onSubmit={[Function]}
  />
</div>
`;
